### PR TITLE
Revise README

### PR DESCRIPTION
**Pull request by shaydewael**
This PR modifies the README to include more information about available listeners and available utility functions. It also does a little bit of reorganizing of the README's order

### Category (place an `x` in each of the `[ ]`)

* [ ] `slack_bolt.App` and/or its core components
* [ ] `slack_bolt.async_app.AsyncApp` and/or its core components
* [ ] Adapters in `slack_bolt.adapter`
* [x] Document pages under `/docs`
* [ ] Others

## Requirements (place an `x` in each `[ ]`)

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_all_and_run_tests.sh` after making the changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
-# Bolt for Python
+# Bolt ![Bolt logo](docs/assets/bolt-logo.svg) for Python (beta)
 
 [![Python Version][python-version]][pypi-url]
 [![pypi package][pypi-image]][pypi-url]
 [![Build Status][travis-image]][travis-url]
 [![Codecov][codecov-image]][codecov-url]
 
-A Python framework to build Slack apps in a flash with the latest platform features. Check the [document](https://slack.dev/bolt-python/) and [examples](https://github.com/slackapi/bolt-python/tree/main/examples) to know how to use this framework.
+A Python framework to build Slack apps in a flash with the latest platform features. Read the [getting started guide](https://slack.dev/bolt-python/tutorial/getting-started) and look at our [code samples](https://github.com/slackapi/bolt-python/tree/main/samples) to learn how to build apps using Bolt.
```
Comment (mwbrooks) [inline README.md:8]: We updated the `samples/` directory to `examples/` to create some consistency across all our SDKs and frameworks: https://github.com/slackapi/bolt-python/tree/main/examples

We can call it "code samples" or "code examples" but using the word "examples" may help hint what folder to look for:

```suggestion
A Python framework to build Slack apps in a flash with the latest platform features. Read the [getting started guide](https://slack.dev/bolt-python/tutorial/getting-started) and look at our [code examples](https://github.com/slackapi/bolt-python/tree/main/examples) to learn how to build apps using Bolt.
```

File: README.md [hunk 2]
```diff
@@ -95,9 +49,53 @@ python app.py
 ngrok http 3000
 ```
 
-## AsyncApp Setup
+## Listening for events
+Apps typically react to a collection of incoming events, which can correspond [Events API events](https://api.slack.com/events-api), [actions](https://api.slack.com/interactivity/components), [shortcuts](https://api.slack.com/interactivity/shortcuts), [slash commands](https://api.slack.com/interactivity/slash-commands) or [options requests](https://api.slack.com/reference/block-kit/block-elements#external_select). For each type of
```
Comment (mwbrooks) [inline README.md:53]: My grammer isn't top notch, but should this be "correspond to"?

```suggestion
Apps typically react to a collection of incoming events, which can correspond to [Events API events](https://api.slack.com/events-api), [actions](https://api.slack.com/interactivity/components), [shortcuts](https://api.slack.com/interactivity/shortcuts), [slash commands](https://api.slack.com/interactivity/slash-commands) or [options requests](https://api.slack.com/reference/block-kit/block-elements#external_select). For each type of
```

**Review by mwbrooks — CHANGES_REQUESTED** [review]
Great update! I really like the list of methods and arguments. Super helpful to anyone learning about Bolt for the first time.

Only feedback is to update the path to the samples directory, which recently changed.

File: README.md [hunk 3]
```diff
@@ -95,9 +49,53 @@ python app.py
...
+request, there's a method to build a listener function.
+
+```python
+// Listen for an event from the Events API
```
Comment (seratch) [inline README.md:57]: how about mentioning decorator first and then having a list of `app.xxx`?

The recommended way is to use Python decorators.

```python
@app.event(event_type)
def handle_event(event):
    pass
```

Here is a list of available methods to register listeners:
```python
app.event(event_type, matcher)
app.message([pattern ,] fn)
```

Please remove `@` from the examples in the list.

File: README.md [hunk 4]
```diff
@@ -95,9 +49,53 @@ python app.py
...
+@app.event(event_type, matcher);
```
Comment (seratch) [inline README.md:58]: remove semicolons in any case

File: README.md [hunk 5]
```diff
@@ -95,9 +49,53 @@ python app.py
...
+@app.action(action_id, fn);
+
+// Listen for dialog submissions
+@app.action({ callback_id: callbackId, block_id: blockId, action_id: actionId }, fn);
+
+// Listen for a global or message shortcuts
+@app.shortcut(callback_id, fn);
+
+// Listen for slash commands
+@app.command(/command_name, fn);
```
Comment (seratch) [inline README.md:73]:
```suggestion
@app.command(command_name, fn);
```

File: README.md [hunk 6]
```diff
@@ -95,9 +49,53 @@ python app.py
...
-## AsyncApp Setup
+## Listening for events
+Apps typically react to a collection of incoming events, which can correspond [Events API events](https://api.slack.com/events-api), [actions](https://api.slack.com/interactivity/components), [shortcuts](https://api.slack.com/interactivity/shortcuts), [slash commands](https://api.slack.com/interactivity/slash-commands) or [options requests](https://api.slack.com/reference/block-kit/block-elements#external_select). For each type of
+request, there's a method to build a listener function.
+
+```python
+// Listen for an event from the Events API
+@app.event(event_type, matcher);
+
+// Convenience method to listen to only `message` events using a string or RegExp
```
Comment (seratch) [inline README.md:60]: There is no `RegExp` in Python. `re.Pattern`.

File: README.md [hunk 7]
```diff
@@ -95,9 +49,53 @@ python app.py
...
+@app.message([pattern ,] fn);
+
+// Listen for an action from a Block Kit element (buttons, select menus, date pickers, etc)
+@app.action(action_id, fn);
+
+// Listen for dialog submissions
+@app.action({ callback_id: callbackId, block_id: blockId, action_id: actionId }, fn);
```
Comment (seratch) [inline README.md:67]: Be consistent to use snake_case - keys in dict must have be `"xxx"`
```suggestion
app.action({ "callback_id": callback_id, "block_id": block_id, "action_id": action_id }, fn);
```

**Review by seratch — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -95,9 +49,61 @@ python app.py
...
+app.event(event_type, fn)
+
+# Convenience method to listen to only `message` events using a string or re.Pattern
+app.message([pattern ,] fn)
+
+# Listen for an action from a Block Kit element (buttons, select menus, date pickers, etc)
+app.action(action_id, fn)
+
+# Listen for dialog submissions
+app.action({ callback_id: callbackId }, fn)
```
Comment (seratch) [inline README.md:67]:
```suggestion
app.action({"callback_id": callbackId}, fn)
```

**Review by seratch — APPROVED** [review]

**Pull request merged**